src/main/java/org/eclipse/xpanse/terraform/boot/terraform/service/TerraformScriptsService.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ *
+ */
+
+package org.eclipse.xpanse.terraform.boot.terraform.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.xpanse.terraform.boot.models.exceptions.TerraformExecutorException;
+import org.eclipse.xpanse.terraform.boot.models.request.TerraformDeployWithScriptsRequest;
+import org.eclipse.xpanse.terraform.boot.models.request.TerraformDestroyWithScriptsRequest;
+import org.eclipse.xpanse.terraform.boot.models.request.async.TerraformAsyncDeployFromDirectoryRequest;
+import org.eclipse.xpanse.terraform.boot.models.request.async.TerraformAsyncDestroyFromDirectoryRequest;
+import org.eclipse.xpanse.terraform.boot.models.response.TerraformResult;
+import org.eclipse.xpanse.terraform.boot.terraform.TerraformExecutor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ *  Terraform service classes are deployed form Scripts.
+ */
+@Slf4j
+@Service
+public class TerraformScriptsService extends TerraformDirectoryService {
+
+    private static final String FILE_SUFFIX = ".tf";
+    private static final String STATE_FILE_NAME = "terraform.tfstate";
+
+    private final RestTemplate restTemplate;
+    private final TerraformExecutor executor;
+
+    /**
+     * TerraformScriptsService constructor.
+     */
+    @Autowired
+    public TerraformScriptsService(TerraformExecutor executor, RestTemplate restTemplate) {
+        super(executor);
+        this.executor = executor;
+        this.restTemplate = restTemplate;
+
+    }
+
+    /**
+     * Method of deployment a service using a script.
+     */
+    public TerraformResult deployWithScripts(TerraformDeployWithScriptsRequest request) {
+        try {
+            buildDeployEnv(request.getTaskId(), request.getScripts());
+            return deployFromDirectory(request, request.getTaskId());
+        } catch (JsonProcessingException e) {
+            log.error("Deployment service error, taskId: {} , error: {}", request.getTaskId(),
+                    e.getMessage());
+            return TerraformResult.builder()
+                    .commandStdOutput(null)
+                    .commandStdError(e.getMessage())
+                    .isCommandSuccessful(false)
+                    .terraformState(null)
```

Comment:
we must also set the state data here also. in case any resources were provisioned in before the deployment failed.